disnake/ext/commands/params.py

```diff
@@ -61,12 +61,17 @@
 from .converter import CONVERTER_MAPPING
 
 if TYPE_CHECKING:
+    from disnake.app_commands import Choices
+    from disnake.types.interactions import ApplicationCommandOptionChoiceValue
+
     from .slash_core import InvokableSlashCommand, SubCommand
 
     AnySlashCommand = Union[InvokableSlashCommand, SubCommand]
 
     from typing_extensions import TypeGuard
 
+    TChoice = TypeVar("TChoice", bound=ApplicationCommandOptionChoiceValue)
```

Comment:
I'm pretty sure this should go outside of the TYPE_CHECKING, and use a string bind, if that's possible.

Reply:
Why should it? It's not being used at runtime anymore, since the unnecessary `cast` got removed

Reply:
Oh wait, yeah, this is fine.